xnmt/decoder.py

```diff
@@ -46,34 +37,21 @@ def __init__(self, rnn_state=None, context=None):
     self.rnn_state = rnn_state
     self.context = context
 
-class MlpSoftmaxDecoder(RnnDecoder, Serializable):
+class MlpSoftmaxDecoder(Decoder, Serializable):
   """
   Standard MLP softmax decoder.
 
   Args:
-    layers (int): number of LSTM layers
     input_dim (int): input dimension
-    lstm_dim (int): LSTM hidden dimension
-    mlp_hidden_dim (int): MLP hidden dimension
     trg_embed_dim (int): dimension of target embeddings
-    dropout (float): dropout probability for LSTM
-    rnn_spec (str): 'lstm' or 'residuallstm'
-    residual_to_output (bool): option passed on if rnn_spec == 'residuallstm'
     input_feeding (bool): whether to activate input feeding
-    param_init_lstm (ParamInitializer): how to initialize LSTM weight matrices (currently, only :class:`xnmt.param_init.GlorotInitializer` is supported)
-    param_init_context (ParamInitializer): how to initialize context weight matrices
-    bias_init_context (ParamInitializer): how to initialize context bias vectors
-    param_init_output (ParamInitializer): how to initialize output weight matrices
-    bias_init_output (ParamInitializer): how to initialize output bias vectors
+    rnn_layer (SeqTransducer): recurrent layer of the decoder; defaults to UniLSTMSeqTransducer
```

Comment:
SeqTransducer is not the correct type here, as SeqTransducers can't consume inputs one by one. For lack of an appropriate abstract class, the requested type should probably be UniLSTMSeqTransducer.

Reply:
Yeah, that's where @neubig's suggestion would probably come in handy. Also, it's not like you should really use the BiLSTMSeqTransducer either, so...